connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java

```diff
@@ -195,10 +210,91 @@ public Connector connector() {
         return connector;
     }
 
+    ConnectorMetricsGroup metrics() {
+        return metrics;
+    }
+
     @Override
     public String toString() {
         return "WorkerConnector{" +
                 "id=" + connName +
                 '}';
     }
+
+    class ConnectorMetricsGroup {
+        private final MetricGroup metricGroup;
+        private final MetricName statusRunningName;
+        private final MetricName statusPausedName;
+        private final MetricName statusFailedName;
+        private final Sensor statusRunning;
+        private final Sensor statusPaused;
+        private final Sensor statusFailed;
+
+        public ConnectorMetricsGroup(ConnectMetrics connectMetrics) {
+            this.metricGroup = connectMetrics.group("connector-metrics",
+                    "connector", connName);
+
+            this.statusRunningName = metricGroup.metricName("status-running",
+                    "Signals whether the connector is in the running state.");
+            this.statusRunning = metricGroup.metrics().sensor("status-running");
+            this.statusRunning.add(statusRunningName, new Value());
+
+            this.statusPausedName = metricGroup.metricName("status-paused",
+                    "Signals whether the connector is in the paused state.");
+            this.statusPaused = metricGroup.metrics().sensor("status-paused");
+            this.statusPaused.add(statusPausedName, new Value());
+
+            this.statusFailedName = metricGroup.metricName("status-failed",
+                    "Signals whether the connector is in the failed state.");
+            this.statusFailed = metricGroup.metrics().sensor("status-failed");
+            this.statusFailed.add(statusFailedName, new Value());
+
+            recordInitialized();
+        }
+
+        public void recordInitialized() {
+            this.statusRunning.record(0);
+            this.statusPaused.record(0);
+            this.statusFailed.record(0);
+        }
+
+        public void recordStopped() {
+            this.statusRunning.record(0);
```

Comment:
huh, i think this was missed during the kip review because of the confusion over how these would be expressed? seems odd that you can't differentiate all the states

Reply:
Yeah, this was missed because I used the `WorkerConnector.State` for the KIP, but once I got into the code I realized that Enum doesn't handle all of the states (including paused). Would appreciate advice as to whether I should add the remaining states and update the KIP, or implement per the KIP.